http/1.7/modules/ROOT/pages/http-polling-source-ref.adoc

```diff
@@ -0,0 +1,300 @@
+
+In the previous simple examples, the server always returned the same HTTP response. In more complex scenarios the server needs to know that it has to send the next response. You can send headers, URI parameters, body, or query parameters, but if these parameters always contain the same values in the HTTP request, the server does not know what would be the next response.
+
+For these polling scenarios, you use watermarking, where the server returns a watermark value on the payload itself, or in every item. For example, the watermark value can be a timestamp that can refer to the entire collection, or every item could have its own timestamp. In any case, you provide a watermark expression using the `watermark` placeholder.
+
+The expression extracts the watermark from the response, and subsequently, use this watermark to send the requests to the server. You can use expressions for the body, headers, URI parameters and query parameters' values in the the `watermark` placeholder.
+
+Note that in the first execution, the `watermark` placeholder value is `null`, which you might want to consider in the server or the expression where the placeholder is used.
+
+* To refer to the entire payload in the watermark expression, use the `payload` placeholder, as in the splitting expression.
+* To refer to an item, use the `item` placeholder, the  watermarking expression applies to every item, one by one.
+
+The following XML configuration example shows an HTTP *Listener* flow. In the first polling iteration, when there is no watermark, a payload is returned with a watermark value set in the `wm` property. In the second polling iteration, a watermark value is expected to be used, so the payload is different. In this case, coming from the request's payload:
+
+[source,xml,linenums]
+----
+<flow name="watermarkInPayloadListenerFlow">
+    <http:listener config-ref="watermarkListenerConfig" path="/watermark-payload"/>
+    <choice>
+        <when expression="#[payload.watermark == null]">
+            <set-payload value="#[output application/json --- {'items': [{'name': 'Eze'}, {'name': 'Fabi'}, {'name': 'Sofi'}], 'wm': 0}]"/>
+        </when>
+        <when expression="#[payload.watermark == '0']">
+            <set-payload value="#[output application/json --- {'items': [{'name': 'Euge'}, {'name': 'Juli'}], 'wm': 1}]"/>
+        </when>
+        <when expression="#[payload.watermark == '1']">
+            <set-payload value="#[output application/json --- {'items': [{'name': 'Pablo'}, {'name': 'Martín'}], 'wm': 2}]"/>
+        </when>
+        <otherwise>
+            <set-payload value="#[output application/json --- {'items': [], 'wm': 3}]"/>
+        </otherwise>
+    </choice>
+</flow>
+----
+
+The following XML configuration example shows how to extract the watermark value from the entire payload, and then use the value in the body of the request. The configuration uses the HTTP *Polling Source* event source instead of the HTTP *Listener* source:
```

Comment:
```suggestion
The following XML configuration example shows how to extract the watermark value from the entire payload and then use the value in the body of the request. The configuration uses the HTTP *Polling Source* event source instead of the HTTP *Listener* source:
```